src/orders/order.service.ts

```diff
@@ -2,31 +2,50 @@ import { Injectable, NotFoundException } from "@nestjs/common";
 import { InjectModel } from "@nestjs/mongoose";
 import { Order, StatusType } from "./schemas/order.schema";
 import { Model, ObjectId, Types } from "mongoose";
-import { FindAllOrderDto } from "./dto/findAllOrder.dto";
+import { FindAllOrderDto, GetOrdersFilterDto } from "./dto/findAllOrder.dto";
 import { CreateOrderDto } from "./dto/createOrder.dto";
 import { UpdateOrderDto } from "./dto/updateOrder.dto";
+import { PaginationQueryDto } from "src/utils/PaginationDto/pagination-query.dto";
 
 @Injectable()
 export class OrderService {
   constructor(@InjectModel(Order.name) private readonly orderModel: Model<Order>) {}
-
-  async findAll(findAllOrder: FindAllOrderDto): Promise<{ data: Order[]; total: number }> {
+  //findAll is for cc-app order, need exact search
+  async findAll(findAllOrder: FindAllOrderDto & PaginationQueryDto): Promise<Order[]> {
     const { user_id, limit = 0, offset = 0 } = findAllOrder;
     if (user_id === "") {
       throw new NotFoundException("userId cannot be empty string");
     }
     const optionalQuery: { [key: string]: any } = {};
     if (user_id) optionalQuery.user_id = user_id;
 
-    const ordersData = await this.orderModel
+    return await this.orderModel
       .find({ ...optionalQuery })
       .populate("paymentInfo")
       .sort({ createdAt: -1 })
       .skip(offset)
       .limit(limit)
       .exec();
+  }
+
+  //find is for admin only
```

Comment:
As status is an optional query, you need to define an optionalQuery, or it will pass "undefined" to find({}) when the frontend does not pass the status. 

//find is for admin only
  async findAllByFilters(
    findAllByfilterDto: GetOrdersFilterDto & PaginationQueryDto,
  ): Promise<{ data: Order[]; total: number }> {
    const { user_id = "", status, limit = 0, offset = 0 } = findAllByfilterDto;
    
    const optionalQuery: { [key: string]: any } = {};
    if (status) optionalQuery["status"] = status;
    const filter = { user_id: { $regex: user_id, $options: "i" }, ...optionalQuery };

    const ordersData = await this.orderModel
      .find(filter)
      .populate("paymentInfo")
      .sort({ createdAt: -1 })
      .skip(offset)
      .limit(limit)
      .exec();

    const total = await this.orderModel.countDocuments(filter);

    return {
      data: ordersData,
      total,
    };
  }

Reply:
user_id is optional query too, so I modify the code as below.

`  async findAllByFilters(
    findAllByfilterDto: GetOrdersFilterDto & PaginationQueryDto,
  ): Promise<{ data: Order[]; total: number }> {
    const { user_id, status, limit = 0, offset = 0 } = findAllByfilterDto;
    const filterQuery : Record<string, any> = {
      ...(user_id ? { user_id: { $regex: user_id, $options: "i" } } : {}),
      ...(status ? { status }: {})
    };
    const ordersData = await this.orderModel
      .find(filterQuery)
      .populate("paymentInfo")
      .sort({ createdAt: -1 })
      .skip(offset)
      .limit(limit)
      .exec();

    const total = await this.orderModel.countDocuments({filterQuery});

    return {
      data: ordersData,
      total,
    };
  }`